### ISSUE TITLE
stitch_multibed(return_stitching_weights=True) returns incorrect weight tensor shape/content for 3D recons

### ISSUE THREAD

**Issue by MengXiangxi** [PyTomography/PyTomography#225]
> Note: This issue was drafted by GitHub Copilot. I encountered the bug while trying to resolve some other problems.
>
> <img width="400" alt="Image" src="https://github.com/user-attachments/assets/81bfefd3-c281-4bec-8872-61e40cb020e0" />

## Summary
It appears that when calling `pytomography.io.SPECT.dicom.stitch_multibed` with `return_stitching_weights=True` on 3D recon inputs shaped `(n_beds, Lx, Ly, Lz)`, the returned `weights` may not preserve an explicit per-bed 3D layout.

In some multi-bed scenarios, this could make downstream diagnostics/uncertainty logic ambiguous or incorrect.

## Environment
- OS: Linux
- Python: 3.13.11
- pytomography: 3.4.0

## Minimal Reproducible Example
```python
import torch
from pytomography.io.SPECT import dicom

# Example only: provide two valid NM DICOM files from different bed positions
files = [
  "/path/to/bed1_projection.dcm",
  "/path/to/bed2_projection.dcm",
]

# 3D recon per bed: (n_beds, Lx, Ly, Lz)
recons = torch.ones((2, 128, 128, 128), dtype=torch.float32)

weights, zs = dicom.stitch_multibed(
  recons,
  files,
  return_stitching_weights=True,
)

print("weights shape:", tuple(weights.shape))
print("zs:", zs)
```

Potentially problematic output pattern (observed in one environment):
- `weights shape == (2, 128, 128)` while input is `(2, 128, 128, 128)`.

## Actual Behavior
- In some cases, `weights` seems to lose or collapse the expected 3D per-bed structure.
- Returned shape/content can be difficult to interpret against a `(n_beds, Lx, Ly, Lz)` input.

## Expected Behavior
- Return per-bed 3D stitching weights consistent with the reconstruction geometry and overlap.
- Shape should be unambiguous/documented (for example an explicit bed axis), without collapsing dimensions unexpectedly.

## Suspected Root Cause
It may be related to this pattern in `stitch_multibed` when `return_stitching_weights=True`:

- `torch.cat(stitching_weights)` merges tensors along `dim=0`.
- Indexing with `np.argsort(order)` is then applied on that first dimension.

For 3D per-bed weights, this could potentially select along a spatial axis rather than a dedicated bed axis, which might explain shape/content mismatch.

## Suggested Fix
One robust approach:
1. Keep bed axis explicit via torch.stack(stitching_weights) to obtain shape (n_beds, Lx, Ly, Lz).
2. Reorder on the bed axis only:
   stitching_weights = torch.stack(stitching_weights)[torch.as_tensor(np.argsort(order), device=...)]
3. Return stitching_weights (plus zs in corresponding order).

Pseudo replacement:
- current:
  return torch.cat(stitching_weights)[np.argsort(order)], zs[np.argsort(order)]
- proposed:
  weights = torch.stack(stitching_weights)
  inv_order = np.argsort(order)
  return weights[inv_order], zs[inv_order]

## Additional Notes
- This report is intentionally data-agnostic and does not include private test files.
- If needed, I can help validate a maintainer-provided synthetic or public test case.

Thanks for maintaining PyTomography.
